src/main/java/thePackmaster/hats/HatMenu.java

```diff
@@ -0,0 +1,162 @@
+package thePackmaster.hats;
+
+import basemod.BaseMod;
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.graphics.Color;
+import com.badlogic.gdx.graphics.g2d.SpriteBatch;
+import com.badlogic.gdx.graphics.g2d.TextureRegion;
+import com.megacrit.cardcrawl.characters.AbstractPlayer;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import com.megacrit.cardcrawl.core.Settings;
+import com.megacrit.cardcrawl.helpers.FontHelper;
+import com.megacrit.cardcrawl.helpers.ImageMaster;
+import com.megacrit.cardcrawl.screens.options.DropdownMenu;
+import thePackmaster.SpireAnniversary5Mod;
+import thePackmaster.ThePackmaster;
+import thePackmaster.packs.AbstractCardPack;
+
+import java.util.ArrayList;
+
+import static thePackmaster.hats.Hats.currentHat;
+
+public class HatMenu {
+
+    public boolean isOpen = false;
+    public static boolean invalidHatSelected = false;
+
+    private static DropdownMenu dropdown;
+    public static final ArrayList<String> hats = new ArrayList<>();
+
+    public static final String[] TEXT = CardCrawlGame.languagePack.getUIString(SpireAnniversary5Mod.makeID("HatMenu")).TEXT;
+    private static final TextureRegion MENU_BG = new TextureRegion(ImageMaster.loadImage("img/ModPanelBg.png"));
+
+    //positions
+    private static final float BG_X_SCALE = Settings.scale * 0.275f;
+    private static final float BG_Y_SCALE = Settings.scale * 0.8f;
+    private static final float BG_X = 525f * Settings.xScale;
+    private static final float BG_Y = Settings.HEIGHT - 40f * Settings.yScale - MENU_BG.getRegionHeight() * BG_Y_SCALE;
+    private static final float DROPDOWN_X = 550f * Settings.xScale;
+    private static final float DROPDOWN_Y = Settings.HEIGHT - 160f * Settings.yScale;
+    private static final float PREVIEW_X = BG_X + (210 * Settings.scale);
+    private static final float PREVIEW_Y = BG_Y + (225 * Settings.scale);
+
+    public static AbstractPlayer dummy;
+
+    private static final boolean UNLOCK_ALL_HATS = true;
```

Comment:
Just leaving a note here to set this to false before merging... and to make sure we test with this set to false 😄.